### PR TITLE
native dom helpers

### PR DESCRIPTION
**Pull request by iezer**

### CODE REVIEW

File: tests/fakes/stripe-checkout.js [hunk 1]
```diff
@@ -23,5 +23,6 @@ export default Service.extend({
         this.set('isClosed', true);
       }
     });
+    return RSVP.resolve(handler);
```
Comment (mixonic) [inline tests/fakes/stripe-checkout.js]: Seems ok. Even better is to use an actual promise wrapper. The reason is that exceptions would be delivered differently.

A method returning a promise should *ALWAYS* return a promise. This includes exceptions. For example using your current logic you would handle an expected exception via:

```js
try {
  serviceInstance.createHandler(cb);
} catch(e) {
  // handle something like `emberObject.create` throwing
}
```

And that is wrong, because what you expect is to use `catch`. Thus you should use a wrapper:

```js
  createHandler(cb) {
    return new RSVP.Promise(resolve => {
      let handler = emberObject.create( // ...
      /* snip */
      resolve(handler);
    });
  },
```

And then you can use `catch`:

```js
serviceInstance.createHandler(cb).catch((e) => {
  // handle something like `emberObject.create` throwing
});
```

You can also then reliably use `await` if the parent function is inside an `async`:

```js
try {
  await serviceInstance.createHandler(cb);
} catch(e) {
  // handle something like `emberObject.create` throwing
});
```

Because of the async timing of promises (catch/then are handled in a microtask, not the current stack) you're also improving the consistency of handling the return value or exception.

**tl;dr**, prefer a wrapper instead of `RSVP.resolve` or `RSVP.reject`.

Comment (iezer) [reply tests/fakes/stripe-checkout.js]: Thanks I'll change it

Comment (iezer) [reply tests/fakes/stripe-checkout.js]: PS @mixonic CI isn't set up for this branch, seems like a simple switch to enable on travis.ci.org but I don't have access. Can you enable?

Comment (mixonic) [reply tests/fakes/stripe-checkout.js]: @iezer should be done. I'm not sure it will pick up this PR though. I triggered a build of master here: https://travis-ci.org/201-created/bodega/builds/339128367
